src/ride/track.c

```diff
@@ -370,7 +370,7 @@ void track_list_populate(ride_list_item item, uint8* track_list_cache){
 		}
 
 		// If cur_track_entry_index is greater than max number of tracks
-		if (cur_track_entry_index >= 1000){
+		if (cur_track_entry_index >= 254){
```

Comment:
Something is wrong then, the original code couldn't have been comparing a byte with a word. This routine will need checking and cleaning up. 1000 is the maximum number of tracks allocated in memory though for a ride's track design list...